fw/ss_skb.c

```diff
@@ -1739,6 +1739,10 @@ ss_skb_cut_extra_data(struct sk_buff *skb_head, struct sk_buff *skb,
 
 			frag_num = -1;
 			skb = skb->next;
+			/* new skb also can be fragmented */
+			if (skb_is_nonlinear(skb))
+				goto frag;
```

Comment:
If we have fragments, this doesn't mean that we can and should skip the linear data, what the change actually does.

#1565 warning apparently raised because we didn't find the stop pointer and just ran out of the fragments. Why this happened? What we have missed? These questions had to be addressed in the commit message. Sine you made this fix, it seems we missed the fragments of the chained skb, but I don't see how that happened. Once we process the linear data with `__split_linear_data()` we're walking over the grags...

Also it seems your patch addresses the problem of an skb with no linear data at all. In this case we have `size == 0` and, correspondingly, `len == 0`, so `__split_linear_data()` is called with zero length. It seems the function doesn't expect such length - see for example that `__skb_fragment()` has `BUG_ON(!len)`. Maybe the fix should be in checking zero length linear data and _only then_ do `goto frag`